maker/src/main.rs

```diff
@@ -50,7 +50,17 @@ async fn main() -> Result<()> {
     let seed = RandomSeed::initialize(&data_dir.join("maker_seed")).await?;
 
     let bitcoin_network = opts.network.bitcoin_network();
-    let ext_priv_key = seed.derive_extended_priv_key(bitcoin_network)?;
+
+    let ext_priv_key = match opts.wallet_xprv {
+        Some(wallet_xprv) => {
+            if wallet_xprv.network != bitcoin_network {
+                let is_network = wallet_xprv.network;
```

Comment:
```suggestion
                let network = wallet_xprv.network;
```

🙃 Reads better in the message below.